src/backend/gporca/libnaucrates/src/init.cpp

```diff
@@ -151,9 +141,17 @@ gpdxl_init()
 //
 //---------------------------------------------------------------------------
 void
-gpdxl_terminate()
+gpdxl_terminate(bool shutdown)
 {
 #ifdef GPOS_DEBUG
+	shutdown = true;
+#endif	// GPOS_DEBUG
+
+	if (!shutdown)
+	{
+		return;
+	}
+
 	ShutdownDXL();
```

Comment:
If malloc returns 0 at `gpos::clib::Malloc (size=304) at clibwrapper.cpp:475`, then pmpXerces is NULL. ShutdownDXL has `GPOS_ASSERT(NULL != pmpXerces)` at the beginning

Reply:
fixed